src/workspaces/workspaces.service.ts

```diff
@@ -70,4 +86,153 @@ export class WorkspacesService {
     }
     return userWorkspace.workspace
   }
+
+  async update(userId:string, workspaceId:string, updateWorkspaceDto: UpdateWorkspaceDto){
+
+
+    const workspace = await this.workspaceRepository.findOne({
+        where: {
+          id: workspaceId
+        }
+    })
+
+    if(!workspace){
+      throw new NotFoundException("Workspace not found")
+    }
+
+    const userWorkspace = await this.userWorkspaceRepository.findOne({
+      where: {
+        userId,
+        workspaceId
+      }
+    })
+
+    if(!userWorkspace) {
+      throw new ForbiddenException('Dear User you are not allowed to edit this workspace')
+    }
+
+
+    await this.workspaceRepository.update({id: workspaceId},{name: updateWorkspaceDto.name})
+    return await this.workspaceRepository.findOne({
```

Comment:
repositories return an updated object after update. There is no need to fetch the workspace again